aten/src/ATen/core/dispatch/Dispatcher.h

```diff
@@ -122,16 +123,16 @@ class CAFFE2_API Dispatcher final {
   void deregisterSchema_(const OperatorHandle& op, const OperatorName& op_name);
   void deregisterBackendFallbackKernel_(TensorTypeId dispatchKey);
 
-  static const KernelFunction& dispatch_(const DispatchTable& dispatchTable, const ska::flat_hash_map<TensorTypeId, KernelFunction>& backendFallbackKernels, c10::optional<TensorTypeId> dispatch_key);
+  const KernelFunction& dispatch_(const DispatchTable& dispatchTable, c10::optional<TensorTypeId> dispatch_key) const;
 
   template<class Return, class... Args>
-  Return doCallUnboxed(const DispatchTable& dispatchTable, const LeftRight<ska::flat_hash_map<TensorTypeId, KernelFunction>>& backendFallbackKernels_, Args... args) const;
+  Return doCallUnboxed(const DispatchTable& dispatchTable, Args... args) const;
   template<class Return, class... Args>
-  Return doCallUnboxedOnly(const DispatchTable& dispatchTable, const LeftRight<ska::flat_hash_map<TensorTypeId, KernelFunction>>& backendFallbackKernels_, Args... args) const;
+  Return doCallUnboxedOnly(const DispatchTable& dispatchTable, Args... args) const;
 
   std::list<OperatorDef> operators_;
   LeftRight<ska::flat_hash_map<OperatorName, OperatorHandle>> operatorLookupTable_;
-  LeftRight<ska::flat_hash_map<TensorTypeId, KernelFunction>> backendFallbackKernels_;
+  ska::flat_hash_map<TensorTypeId, KernelFunction> backendFallbackKernels_;
```

Comment:
this should also be changed to an array for the same concurrency concerns we had with the dispatch table. Maybe it actually makes sense to factor out the concept of a `KernelTable` (but up to you)